.github/workflows/ci.yaml

```diff
@@ -148,6 +150,11 @@ jobs:
             hashFiles('requirements.txt') }}-${{
             hashFiles('requirements_test.txt') }}-${{
             hashFiles('homeassistant/package_constraints.txt') }}"
+      - name: Generate partial pip restore key
```

Comment:
This should not be needed, as the setup Python action can take care of that directly

Reply:
It could, but doing it separately gives us a bit more control. It's for example not necessary to restore the cache if we don't need to recreate the venv. That saves about 20-30s on a normal run (per job).

_The date in the cache key is just used to create a unique key each time. The action will then always fall back to restore the last cache entry instead and, since no direct cache hit occurred, it will store the new one._

Reply:
> It could, but doing it separately gives us a bit more control.

ok

> The date in the cache key is just used to create a unique key each time

This does not make sense to me, the cache is stored at the end of the run anyways, regardless if the key changed or not.

Reply:
> > The date in the cache key is just used to create a unique key each time
> 
> This does not make sense to me, the cache is stored at the end of the run anyways, regardless if the key changed or not.

I might be missing something, but from my testing I found that it's only stored if no exact cache hit occurred, i.e. it needed to use one of the fallback keys.

That does make somewhat sense. It's fair to assume the user doesn't need it to be stored again if an exact hit happened. In the end that's the majority of runs and if the file hashes didn't change, it usually means no new requirements were installed.

Here I'm just using the date instead of file hashes.

Reply:
So why not file hashes?

Reply:
Tbh I wasn't sure which cache entry would be restored. Since I always wanted the last one, I though of datetime. As it works, I didn't change it back. There isn't a performance gain in using the file hashes either as the cache is only ever restored if the key (and by extend the hashes) for the main venv didn't match.

Reply:
Yeah I guess it doesn't matter that much 👍 

Thanks!